core/controllers/resources_test.py

```diff
@@ -495,9 +495,9 @@ def test_request_invalid_asset_type(self) -> None:
         """Test that requests for invalid asset type is rejected with a 400."""
```

Comment:
Does this docstring need to be updated?

Also, why did this change? Is it because the passed value doesn't match the schema (do we always throw 404 rather than 400 in such cases)?

Reply:
> Does this docstring need to be updated?

Yes, updated!

> why did this change? Is it because the passed value doesn't match the schema (do we always throw 404 rather than 400 in such cases)?

Previously, I was using this `assets/<asset_type>/<encoded_filename>` URL so invalid asset_type able to reach the schema_validation and throw 400 InvalidInputException. But when we use `assets/<asset_type:(image|audio|thumbnail)>/<encoded_filename>` URL, then exception for invalid asset_type is raised before it reaches the schema validation because in url itself we are define limited allowed values so exception status changed to 404.